conf/layer.conf

```diff
@@ -16,6 +16,7 @@ BBFILE_PRIORITY_webkit = "7"
 LAYERDEPENDS_webkit = "\
     core \
     openembedded-layer \
+    webserver \
 "
```

Comment:
I think this change is not right, meta-webserver is part of meta-openembedded and were are not specifiying there all the other meta-openembedded sublayers that we depend on

Reply:
`RDEPENDS:packagegroup-wpewebkit-depends-tests` adds `apache2` (at `packagegroup-wpewebkit-depends.bb`)

Without that patch, a world build will fail (or any other explicit build that includes `packagegroup-wpewebkit-depends`)

Reply:
so either that should be removed (as it looks rather test specific) or the meta-webserver will need to be mandatory

Reply:
It is added for the execution of the layout tests. The question is whether the image is being used for this or not in the WebKit CI (or is expected).

Reply:
I see both point of views. I  think the observation that @kwavnet  is right however the proposal is too much restrictive and forces to meta-webkit a dependency that is only required for development what is not ideal for other case of uses (more common cases). 

I think we are OK with the https://github.com/Igalia/meta-webkit/pull/521/commits/3bfa4775394a2c02d460d08283723f00277778db  change. So, first of all I'd suggest to split the proposal into two PRs:

- One exclusively for the changes related to licenses.
- Another for implementing a dynamic logic to handle `meta-webserver` and other dependencies with other layers.


then, **this is my proposal for the second commit:**

To address the issue of handling optional dependencies like `meta-webserver` and avoiding forcing its inclusion in `bblayers.conf` as a hard dependency, I propose implementing an approach based on **dynamic layers**. This would ensure that features related to web servers, such as Apache, are activated only if the `meta-webserver` layer is available:

**Dynamic Layer Logic:**
   - Create a conditional `meta-webserver/recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bbappend`. There we will move  the https://github.com/Igalia/meta-webkit/blob/main/recipes-browser/packagegroups/packagegroup-wpewebkit-depends.bb#L262
   
      ```
      RDEPENDS:packagegroup-wpewebkit-depends-tests:appends = " apache2 apache2-scripts" 
      ```
   - So, if the layer is available, include the related `webserver` packages (e.g., Apache) in the package group.

This proposal has the following **benefits:**
   - Avoids requiring users to add `meta-webserver` to `bblayers.conf` if they don’t need it.
   - Keeps the layer more modular and reduces unnecessary strict dependencies.
   - We could do a similar change for  other optional layers like `meta-multimedia`.

Finally I think it is also important a clear documentation about this behavior in the README files or related commits.

Reply:
Well, from my point of view.
meta-openembedded/meta-oe is mandatory already, hence adding another layer checked out from the same repository doesn't hurt right now.

But what I totally agree on, is that the entire `packagegroup-wpewebkit-depends.bb` file needs to be rethought.
Too much stuff is cramped in there (and I can't say from my limited point of view what each of these dependencies are for).

So I would like to make a counter proposal.

Everything related to tests is moved to a sublayer e.g. meta-webkit-test, that sets those requirements.
That layer can be added to the mix (e.g. here in CI) if needed, but for consumers (such as myself) it remains optional.

An example of that is https://github.com/meta-flutter/meta-flutter and the sublayer https://github.com/meta-flutter/meta-flutter/tree/scarthgap/meta-flutter-apps.

My go to approach would be to merge this here and do the split in the next step.
I'm happy to help but I would need some input what out of the monolithic `packagegroup-wpewebkit-depends.bb` is truly optional, resp. what is test related

Reply:
Alright, commit removed

Reply:
Thanks! merged. Feel free to send another PR for the other change if you wish.

Reply:
Another solution it could be to use  [LAYERRECOMMENDS](https://docs.yoctoproject.org/dev/ref-manual/variables.html#term-LAYERRECOMMENDS)  in combination with  anonymous functions like:

```
python () {
    if d.getVar('BBLAYERS').find('meta-webserver') != -1:
        d.appendVar('RDEPENDS:packagegroup-wpewebkit', ' apache2')
}
```